docs/proposals/formatted-output.md

```diff
@@ -0,0 +1,322 @@
+  - [oras discover](#oras-discover)
+- [FAQ](#faq)
+
+## Background
+
+ORAS has prettified output designed for humans. However, for machine processing, especially in some automation scenarios like scripting and CI/CD pipelines, developers want to perform batch operations and chain different commands with ORAS, as well as filtering, modifying, and sorting objects based on the outputs that are emitted by the ORAS command. Developers expect the ORAS output can be generated as machine-readable text not only the prettified or tabular data, so that they can use the formatted outputs like JSON format to perform further advanced operations. 
+
+The formatted output is not intended to supersede the prettified human-readable and friendly output text of ORAS CLI. It aims to provide programming-friendly experience for developers who want to automate their workflows on machines especially on Unix and most Unix-like operating systems, without parsing unstructured text. It will increase the developer experience for ORAS in automation and scripting scenarios. 
+
+## Guidance
+
+Provide two major options to enable users to define the output format of ORAS commands:
+
+- Use `--output` to output a file or directory in the filesystem
+- Use `--format` to format the output of ORAS commands into different data formats or enable process output data using the given Go template.
+  - Use `--format json|tree|table` to print the output in prettified JSON, tree view, or table view
+  - Use `--format '{{ GO_TEMPLATE_FUNCTION }}'` to enable extract or compose the output data using Go template functions
+  - If user doesn't specify `--format` flag, the default output should be raw JSON data 
+
+See sample use cases of formatted output for `oras manifest fetch`:
+
+- Use `--output` to output a manifest file in the filesystem:
+
+```bash
+oras manifest fetch $REGISTRY/$REPO:$TAG --output sample-manifest.json
+```
+
+View the generated manifest file, it should be raw JSON data
+
+```
+$ cat sample-manifest.json
+{"schemaVersion":2,"mediaType":"application/vnd.oci.image.manifest.v1+json","artifactType":"application/vnd.unknown.artifact.v1","config":{"mediaType":"application/vnd.oci.empty.v1+json","digest":"sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a","size":2,"data":"e30="},"layers":[{"mediaType":"application/vnd.oci.image.layer.v1.tar","digest":"sha256:6cb759c4296e67e35b0367f3c0f51dfdb776a0c99a45f39d0476e43d82696d65","size":14477,"annotations":{"org.opencontainers.image.title":"sbom.spdx"}},{"mediaType":"application/vnd.oci.image.layer.v1.tar","digest":"sha256:54c0e84503c8790e03afe34bfc05a5ce45c933430cfd9c5f8a99d2c89f1f1b69","size":6639,"annotations":{"org.opencontainers.image.title":"scan-test-verify-image.json"}}],"annotations":{"org.opencontainers.image.created":"2023-12-15T09:41:54Z"}}
+```
+
+- Use `--format json` to print the output in prettified JSON:
+
+```bash
+oras manifest fetch $REGISTRY/$REPO:$TAG --format json
```

Comment:
What will happen if you have `--output` and `--format` at the same time?
```
oras manifest fetch $REGISTRY/$REPO:$TAG --output sample-manifest.json --format json
```

Reply:
I like this question. 
We can define that if --format json is specified the output should go to the stdout as json. 
But the file will also be create. Does that sound reasonable?

Reply:
IMHO, the desired UX should be similar to what we have now with `--descriptor`: `--format` and `--output -`(output manifest content to stdout) cannot be used at the same time

Reply:
There is no conflicts when using `--output` and `--format` at the same time. In this case, a manifest file should be produced in the filesystem and the prettified JSON should be outputted on the console:. I updated the example to include this case at line36.